store/root.js

```diff
@@ -83,6 +84,33 @@ class Root {
     const { data } = await apiRequest(url, 'GET', {}, {}, true).promise
     Object.assign(this.statistics, data)
   }
+
+  @action
+  async updateDataProvider(patch) {
+    setTimeout(() => {
+      Object.assign(this.dataProvider, patch)
+    }, 300)
+    return
+
+    // TODO: Should call API
+    // eslint-disable-next-line no-unreachable
+    try {
+      // TODO: Should be this.dataProvider.url
+      const url = `/data-providers/${this.dataProvider.id}`
+      const { data } = await apiRequest(url, 'PATCH', {}, {}, patch)
+      Object.assign(this.dataProvider, data)
+    } catch (networkOrAccessError) {
+      // Ignore errors for this moment
```

Comment:
How to handle errors is quite a big question for this moment. I do not see it in the current implementation. Although, we may add some error field in the store and push errors there. In the components we may have an effect hook on errors.

How would you do it?

Reply:
It depends on how you want to display them. We can handle errors in the global store and display notification via a portal or we can handle every error individually in a component.

Reply:
I would like to have a notification system. It would be easier to handle. However, there are some edge cases which should be handled:

- when error happens, does not matter where it's shown, the component should refresh
- it's better to show contextual errors in the context